apps/playground/src/app/quiz/containers/dependency-injection-quiz/dependency-injection-quiz.component.ts

```diff
@@ -0,0 +1,285 @@
+import { Component, ElementRef, OnInit, Output, ViewChild } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+
+import { QuizQuestion } from '../../model/QuizQuestion';
+
+@Component({
+  selector: 'codelab-dependency-injection-quiz-component',
+  templateUrl: './dependency-injection-quiz.component.html',
+  styleUrls: ['./dependency-injection-quiz.component.scss']
+})
+export class CodelabDependencyInjectionQuizComponent implements OnInit {
+  @Output() question: QuizQuestion;
+  @ViewChild("questionElem") questionElem: ElementRef;
+  answer: number;
+
+  totalQuestions: number;
+  completionTime: number;
+  correctAnswersCount = 0;
+  percentage: number;
+
+  currentQuestion = 1;
+  questionIndex: number;
+  badgeQuestionNumber: number;
+  correctAnswer: boolean;
+  @Output() hasAnswer: boolean;
+  answered: boolean;
+  disabled: boolean;
+  quizIsOver: boolean;
+  progressValue: number;
+  timeLeft: number;
+  timePerQuestion = 20;
+  interval: any;
+  elapsedTime = 0;
+  elapsedTimes = [];
+  finalAnswers = [];
+  showExplanation: boolean;
+
+  allQuestions: QuizQuestion[] = [
+    {
+      index: 1,
+      questionText: 'What is the objective of dependency injection?',
+      options: [
+        { optionValue: 1, text: 'Pass the service to the client.' },

```

Comment:
options: [
        { optionValue: 1, text: 'Pass the service to the client.' },
 

Multi-line suggestions are here!

You can now suggest code changes to multiple lines. Learn more.

 
correct: true

Attach files by dragging & dropping, selecting or pasting them.
        {  text: 'Allow the client to find service.' ,  correct: true},
        {  text: 'Allow the client to build service.' },
        {  text: 'Give the client part service.' }
      ],